File: Java-Kotlin/src/main/kotlin/matrix/Reshape the Matrix.kt
package matrix


/**
 * https://leetcode.com/problems/reshape-the-matrix/
 * */

fun matrixReshape(mat: Array<IntArray>, r: Int, c: Int): Array<IntArray> {
    val n = mat.size
    val m = mat[0].size
    if (n * m != r * c) return mat
    val ans = Array(r) { IntArray(c) }
    for (i in mat.indices)
        for (j in mat[0].indices) {
            val index = i * m + j
            ans[index / c][index % c] = mat[i][j]
        }
    return ans
}

File: C++/Array/Reshape the Matrix.cpp
#include <bits/stdc++.h>

using namespace std;

/**
 * https://leetcode.com/problems/reshape-the-matrix/
 * */

class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>> &mat, int r, int c) {
        auto n = mat.size(), m = mat[0].size();
        if (n * m != r * c) return mat;
        vector<vector<int>> result(r, vector<int>(c));
        for (int i = 0; i < n; i++)
            for (int j = 0; j < m; j++) {
                int index = i * m + j;
                result[index / c][index % c] = mat[i][j];
            }
        return result;
    }
};
